### ISSUE TITLE
DEP0079 warning on Node 10

### ISSUE THREAD

**Issue by muuvmuuv** [Automattic/mongoose#6420]
I get following error: `(node:30514) [DEP0079] DeprecationWarning: Custom inspection function on Objects via .inspect() is deprecated` when I use:

```
Games.find({ $text: { $search: T } },
    (err, docs) => {
      if (docs) {
        console.log("Games:", docs);
        deferred.resolve({
          games: docs
        });
      } else {
        deferred.reject(new Error(err.message));
      }
    }
  );
```
`T` is just a text string like "I search for some food". The error disappears when I use a $or operator with some $elemMatch cases. I really need to use full-text search, my script breaks the hole function when this error occurs.

Node: 10.0.0
mongoose: 5.0.17
Entry from games collection: https://hastebin.com/nonutobeva.json

**Comment by vkarpov15** [comment]
@muuvmuuv this isn't an error, just a deprecation warning in Node.js 10. Looks like [this](https://nodejs.org/api/deprecations.html#deprecations_dep0079_custom_inspection_function_on_objects_via_inspect) is the relevant deprecation. Mongoose relies on custom `inspect()` functions to make debug output readable, but looks like Node.js 12.x will not support that. We'll work around that, but for now it is safe to ignore this warning.

**Comment by medve-dev** [comment]
DEP0079

**Comment by yelworc** [comment]
I might try working on this. If it requires a deep understanding of Mongoose internals, I'm probably not the right guy, though :)

At first glance it looks pretty straightforward, though. The canonical replacement for custom `inspect` methods seems to be the [`util.inspect.custom`](https://nodejs.org/api/util.html#util_custom_inspection_functions_on_objects) symbol.

Thing is, symbols are only available in Node >=6, and Mongoose is still targeting Node 4. I guess a fix for this would require dropping support for 4 and 5 (which should be fine since they are both EOL), and could therefore only be included in the next major Mongoose version, right?

**Comment by vkarpov15** [comment]
Yep you're right re: mongoose supporting node v4 and v5. We'll drop v4 and v5 support in our next major release, but given that we just released 5.0.0 a few months ago, I'd rather not make another major release for the next several months.

**Comment by setsenbat** [comment]
I had the same issue when tried to run loop inside of request:
`
server.post('/count', (req, res, next) => {
        //Check for JSON
        if (!req.is('application/json')) {
            return next(new errors.InvalidContentError("Expects 'application/json'"));
        }
        let counts = req.body;
        var element = Object.keys(counts);

        const count = new Count();
        var epcs = count._EPC;

        element.forEach(async (e) => {
            // console.log(counts[e]._EPC);
            epcs.push({
                _EPC: counts[e]._EPC
            });
            
            const x = await Inventory.find({});
            console.log(x); 
            const newUpdate = await x.forEach(element => {
                 let epc1 = element.epc;
                 console.log(epc1);
             });

            try {
                const newCount = await count.save();
                await res.send(201);
                next();
            } catch (err) {
                return next(new errors.InternalError(err.message));
            }
        });
    });`
